app/javascript/lib/scp-api-metrics.js

```diff
@@ -144,11 +146,38 @@ export function logFilterSearch(facet, terms) {
   )
 }
 
+/**
+ * Logs time between user action and its last interactive effect.
+ *
+ * This provides a higher-level view of timing information also available
+ * in more granular events, but in manner that's easier to find and less
+ * implementation-specific.  See also: logPlot in javascripts/application.js.
+ */
+export function logUserAction(lastEvent, perfTime) {
+  var isScatter = lastEvent === 'plot:scatter';
+  var pageName = window.SCP.analyticsPageName;
+  var isStudyOverview = pageName === 'site-study';
+
+  // Consider using this construct more widely
+  var isFullyInteractive = typeof window.SCP.fullyInteractive !== 'undefined';
+
+  if (isScatter && !isFullyInteractive && isStudyOverview) {
+    log(`user-action:page:view:${pageName}`, {perfTime});
+    window.SCP.fullyInteractive = true;
+  }
+
+  var isGeneSearchEffect = lastEvent.includes('plot') && !isScatter;
+  if (isGeneSearchEffect && isStudyOverview) {
+    log(`user-action:search:${pageName}`, {perfTime});
+  }
```

Comment:
hmmm... it seems like this function requires a lot of hard-coding event names and logic.  Let's find a time to brainstorm other possible approaches

Reply:
Helpful chat!  Per that, I'll wait to review your upcoming refinement that preserves functionality diagrammed above.

Reply:
Thanks for addressing this in #718 -- fantastic approach.